src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,306 @@
+ * - The component has a wrapping `<nav>` with `aria-label="pagination"` included
+ * - Links inside of a `<ul>` list
+ * - Includes `aria-current="page"` on the current page link
+ */
+@Component({
+  tag: 'va-pagination',
+  styleUrl: 'va-pagination.css',
+  scoped: true,
+})
+export class VaPagination {
+  /**
+   * This event is fired when a pagination link is selected
+   */
+  @Event({
+    bubbles: true
+  })
+  pageSelect: EventEmitter;
+
+  /**
+   * The current page number
+   */
+  @Prop({ reflect: true }) page: number = 1;
+  /**
+   * The total number of pages
+   */
+  @Prop() total: number = 1;
+  /**
+   * Number of pages to show on the edges; maxes out at 3
+   */
+  @Prop() edges: number = 1;
+  /**
+   * The maximum number of pages to show at once. This includes the 2 edges & 2
+   * ellipsis even when not visible. Best if this is an odd number to maintain
+   * symmetry
+   */
+   @Prop() maxVisible: number = 9;
+  /**
+   * Text added to aria-labels. Any included `{page}` will be replaced with the
+   * associated page number
+   */
+  @Prop() linkAriaLabel: string = 'Load page {page}';
+
+  @State() currentPage: number = this.page;
+
+  /**
+   * Clamped props passed to functions
+   * @typedef Options
+   * @property {number} current - current page
+   * @property {number} total - total pages
+   * @property {number} edges - pages around first & last page
+   * @property {number} visible - max number of visible pages
+   */
+  /**
+   * Validate settings & return clamped results
+   * @returns Options
+   */
+  private checkValidation = () => {
+    const opts = {
+      current: this.currentPage < 1 ? 1 : this.currentPage,
+      total: this.total < 1 ? 1 : this.total,
+      edges: this.edges < 0 ? 0 : this.edges,
+      visible: this.maxVisible,
+    }
+    if (opts.current > opts.total) {
+      opts.current = opts.total;
+    }
+    // Set to a max of 3 to match the default of 9 maxVisible. We probably
+    // wouldn't want to show more than 3 edge pages anyway. It would get really
+    // messy on small screens
+    if (opts.edges > 3) {
+      opts.edges = 3;
+    }
+    // edges * 2 + 2 ellipsis + 1 visible active link in the middle
+    let minVisible = opts.edges * 2 + 2 + 1;
+    if (minVisible > opts.total) {
+      minVisible = opts.total;
+    }
+    if (minVisible > opts.visible) {
+      opts.visible = minVisible;
+    }
+    return opts;
+  }
+
+  /**
+   * Set current page state & emit pageSelect event
+   * @param page - user selected page
+   */
+  private onPageSelected = (page: number) => {
+    this.currentPage = page;
+    this.pageSelect.emit(page);
+  };
+
+  /**
+   * Key down event handler - this allows the user to hold down the key to
+   * quickly navigate through the pages
+   * @param key - event.key
+   * @param page - page associated with the interaction
+   */
+  private handleKeyDown = (key: string, page: number) => {
+    const activate = ['Enter', ' ', 'Spacebar' /* IE11 */];
```

Comment:
I'm going to remove this based on this article - https://www.a11yproject.com/posts/2019-02-15-creating-valid-and-accessible-links/#toc_Focus-state-and-keyboard